cadc-registry/src/main/java/ca/nrc/cadc/reg/CapabilitiesWriter.java

```diff
@@ -143,12 +146,37 @@ private Element getCapabilityElement(Capability c) {
         
         // extensions
         if (ext) {
-            ret.addContent(c.getExtensionMetadata());
+            for (Element e : c.getExtensionMetadata()) {
+                ret.addContent(deepCopy(e));
+            }
         }
         
         return ret;
     }
     
+    private Attribute deepCopy(Attribute a, Namespace ns) {
+        return new Attribute(a.getName(), a.getValue(), ns);
+    }
+    
+    private Element deepCopy(Element e) {
+        if (!Namespace.NO_NAMESPACE.equals(e.getNamespace())) {
+            throw new UnsupportedOperationException("expected extension metadata to be in " + Namespace.NO_NAMESPACE + ", found: " + e.getNamespace());
+        }
+        
+        Element ret = new Element(e.getName(), Namespace.NO_NAMESPACE);
+        for (Attribute a : e.getAttributes()) {
+            ret.setAttribute(deepCopy(a, Namespace.NO_NAMESPACE));
+        }
+        if (e.getChildren().isEmpty()) {
+            ret.setText(e.getText());
+        } else {
+            for (Element c : e.getChildren()) {
```

Comment:
Is there no text when there are children?

Reply:
correct